### DISCUSSION TITLE
Brew and MacPorts - coexistence possible?

### DISCUSSION THREAD

**Discussion by LostAccount** [Homebrew/discussions#1779] [Everyday usage] [answered]
Hello

I am stuck on Mac OS High Sierra (10.13.6) and currently use brew to manage packages.

I would like to install MacPorts but I understand that this is not generally advised. Could anyone shed some light on this concern please.

Thanks

**Comment by carlocab** [comment, marked as answer]
Having both of them installed used to cause users problems because this would often cause builds to fail. This usually happens when a `brew` build is confused by MacPorts-installed software, or a `port` build gets confused by Homebrew-installed software.

These days, both `brew` and `port` try to avoid interfering with each other's installs, so you might be able to manage having both installed. There are no guarantees that this works as intended, however.

**Comment by tjluoma** [comment]
I have a Mac that cannot upgrade past High Sierra.

I made a list from `brew leaves`, uninstalled `brew` and installed all of the items from the list in MacPorts. Seemed to work fine.

**Comment by LostAccount** [comment]
Thank you both

**Comment by kencu** [comment]
pretty much everything installed by macports is in /opt, so it is off the usual search path for compilers and linkers. You must specifically add the include and library paths to find and use it. This isolation is both a blessing and a hassle, as build systems need to be told where to find it, and many people don't have the skills to manage that step.

Most things installed by homebrew are in /usr/local, so they are on the automatic search path built in to compilers and linkers, and are found and used whether you intended them to be or not. This is also both a blessing and a hassle, of the opposite kind. Things are found and used when you may not have specifically wanted them to be used.

To avoid macports when using homebrew, make sure /opt/local/ is not in your PATH, and you're done.

To avoid homebrew when using macports, you have to move /usr/local somewhere else, like rename it to /usr/local-disabled. Then the compiler and linker won't find anything there.

To build things with macports when homebrew has installed software in /usr/local is a recipe for baldness, as you will likely be finding and using software in /usr/local that macports knows nothing about.

**Reply by tjluoma** [reply]
> Most things installed by homebrew are in /usr/local, so they are on the automatic search path built in to compilers and linkers, and are found and used whether you intended them to be or not. This is also both a blessing and a hassle, of the opposite kind. Things are found and used when you may not have specifically wanted them to be used.

For M1 Macs, most things are in `/opt/homebrew/` which means that, eventually, almost anything `brew` installs will not be in `/usr/local/`.
